docs/code/EntryPoints.md

```diff
@@ -0,0 +1,295 @@
+﻿# Entry Points And Helper Classes 
+
+## Overview
+
+An 'entry point', is a representation of a ML.NET type in JSON format. Entry points are used to serialize and deserialize an ML.NET type in JSON. 
```

Comment:
> An 'entry point', is a representation of a ML.NET type in JSON format.

I am not entirely enthusiastic about that description. I think the primary reason why I don't like it is, I think the phrase ML.NET type is misleading, or at least vague. If I were asked what an ML.NET type is I might say something like `VBuffer` or `IDataView`, and to me a representation as JSON makes me think that thing is being serialized, which is not the point of entry-points at all.

So maybe, we could replace a lot of this language with something like this (I don't insist on this exact wording):

> Entry-points are a way to interface with ML.NET components, by specifying an execution graph of connected inputs and outputs of those components. Both the manifest describing available components and their inputs/outputs, and an "experiment" graph description, are expressed in JSON. Etc. Etc.